spring-integration-ftp/src/test/java/org/springframework/integration/ftp/inbound/RotatingServersTests.java

```diff
@@ -0,0 +1,299 @@
+		assertThat(config.latch.await(10, TimeUnit.SECONDS)).isTrue();
+		List<Integer> sfCalls = config.sessionSources.stream().limit(17).collect(Collectors.toList());
+		assertThat(sfCalls).containsExactly(1, 1, 1, 2, 2, 2, 3, 3, 3, 1, 1, 2, 2, 3, 3, 1, 1);
+		File f1 = new File(tmpDir + File.separator + "f1");
+		assertThat(f1.exists()).isTrue();
+		File f2 = new File(tmpDir + File.separator + "f2");
+		assertThat(f2.exists()).isTrue();
+		File f3 = new File(tmpDir + File.separator + "f3");
+		assertThat(f3.exists()).isTrue();
+		assertThat(f1.delete()).isTrue();
+		assertThat(f2.delete()).isTrue();
+		assertThat(f3.delete()).isTrue();
+		ctx.getBean("files", QueueChannel.class);
+		assertThat(ctx.getBean("files", QueueChannel.class).getQueueSize()).isEqualTo(3);
+		ctx.close();
+	}
+
+	@Test
+	public void testFair() throws Exception {
+		AnnotationConfigApplicationContext ctx = new AnnotationConfigApplicationContext(FairConfig.class);
+		StandardConfig config = ctx.getBean(StandardConfig.class);
+		assertThat(config.latch.await(10, TimeUnit.SECONDS)).isTrue();
+		List<Integer> sfCalls = config.sessionSources.stream().limit(17).collect(Collectors.toList());
+		assertThat(sfCalls).containsExactly(1, 1, 2, 2, 3, 3, 1, 1, 2, 2, 3, 3, 1, 1, 2, 2, 3);
+		File f1 = new File(tmpDir + File.separator + "f1");
+		assertThat(f1.exists()).isTrue();
+		File f2 = new File(tmpDir + File.separator + "f2");
+		assertThat(f2.exists()).isTrue();
+		File f3 = new File(tmpDir + File.separator + "f3");
+		assertThat(f3.exists()).isTrue();
+		assertThat(f1.delete()).isTrue();
+		assertThat(f2.delete()).isTrue();
+		assertThat(f3.delete()).isTrue();
+		assertThat(ctx.getBean("files", QueueChannel.class).getQueueSize()).isEqualTo(3);
+		ctx.close();
+	}
+
+	@Test
+	public void testVariableLocalDir() throws Exception {
+		AnnotationConfigApplicationContext ctx = new AnnotationConfigApplicationContext(VariableLocalConfig.class);
+		StandardConfig config = ctx.getBean(StandardConfig.class);
+		assertThat(config.latch.await(10, TimeUnit.SECONDS)).isTrue();
+		List<Integer> sfCalls = config.sessionSources.stream().limit(17).collect(Collectors.toList());
+		assertThat(sfCalls).containsExactly(1, 1, 1, 2, 2, 2, 3, 3, 3, 1, 1, 2, 2, 3, 3, 1, 1);
+		File f1 = new File(tmpDir + File.separator + "foo" + File.separator + "f1");
+		assertThat(f1.exists()).isTrue();
+		File f2 = new File(tmpDir + File.separator + "baz" + File.separator + "f2");
+		assertThat(f2.exists()).isTrue();
+		File f3 = new File(tmpDir + File.separator + "fiz" + File.separator + "f3");
+		assertThat(f3.exists()).isTrue();
+		assertThat(f1.delete()).isTrue();
+		assertThat(f2.delete()).isTrue();
+		assertThat(f3.delete()).isTrue();
+		assertThat(f1.getParentFile().delete()).isTrue();
+		assertThat(f2.getParentFile().delete()).isTrue();
+		assertThat(f3.getParentFile().delete()).isTrue();
+		assertThat(ctx.getBean("files", QueueChannel.class).getQueueSize()).isEqualTo(3);
```

Comment:
The test fails for me on the regular basis:
```
org.junit.ComparisonFailure: 
Expected :3
Actual   :4
 <Click to see difference>


	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at org.springframework.integration.ftp.inbound.RotatingServersTests.testVariableLocalDir(RotatingServersTests.java:144)
```

Reply:
Looks like we need to clean up a `tmpDir` before each test, e.g. using a `recursiveDelete()` from the super `RemoteFileTestSupport`